### ISSUE TITLE
Where does your script download installers to while downloading?

### ISSUE THREAD

**Issue by LeeBinder** [munki/macadmin-scripts#82]
Hi munki. Thank you for your very handy script. Can you please share the file system path where it temporarily saves downloads at _**while downloading**_? Best in the ReadMe? It's not the usual Apple Software Update `/private/var/folders`/ or `/Library/Updates/`

**Comment by gregneagle** [comment]
```
$ ./installinstallmacos.py --help
usage: installinstallmacos.py [-h] [--seedprogram SEEDPROGRAM]
                              [--catalogurl CATALOGURL]
                              [--workdir path_to_working_dir] [--compress]
                              [--raw] [--ignore-cache]

optional arguments:
  -h, --help            show this help message and exit
      <...snip...>
 --workdir path_to_working_dir
                        Path to working directory on a volume with over 10G of
                        available space. Defaults to current working
                        directory.
     <...snip...>
```

Files are downloaded to the current working directory or the directory given to `--workdir`. The Apple Software Update process is not involved here.

**Comment by LeeBinder** [comment]
OK.. thanks for the -h snippet.. but still, I'm not really any further now. Do you mean macOS' default "current working directory", or the one your script uses "by default"? The `content` folder did not fill up until the download was finished. I couldn't see _**any**_ volume fill up during the download. Written into memory and/or swap file?

If you don't know yourself: do you have a terminal command to display the "current working directory"?

**Comment by gregneagle** [comment]
`pwd`
The current working directory is a common UNIX concept: it's the directory you are working in!
If you are asking "where does `curl` cache/stage downloads as its downloading?" -- I don't know and haven't had a reason to need to find out. What, exactly, is the problem you are trying to solve?

**Comment by gregneagle** [comment]
```
bash-3.2$ cd /Users/Shared
bash-3.2$ pwd
/Users/Shared
bash-3.2$ cd /tmp
bash-3.2$ pwd
/tmp
```
`pwd` stands for "print working directory".

**Comment by LeeBinder** [comment]
yes, thank you. I copied your script to the volume I wanted it to download to. But during the download I realized that the content folder did NOT fill up at all, not even when the dl was already at several gigs. And even odder, I was not able to locate which volume it was filling up. I did also manually look into `/tmp` but not into `/Users/Shared` - but `/` itself did not fill up..

I in general like to have control over large downloads. You write your script uses curl, so it must be curl determining the temp download folder location. Maybe newer versions indeed rather cache into memory or page file vm, which I think is hidden in the APFS container.

Anyway, at this point I'm fine accepting a little mystery :)
